src/main/java/org/retest/rebazer/service/BitbucketService.java

```diff
@@ -39,21 +39,23 @@ public BitbucketService(RebaseService rebaseService) {
 
 	@Scheduled(fixedDelay = 60 * 1000)
 	public void pollBitbucket() {
-		final List<PullRequest> allPullRequests = getAllPullRequestIds();
-
-		for (final PullRequest pullRequest : allPullRequests) {
-			log.debug("processing " + pullRequest);
-
-			if (!greenBuildExists(pullRequest)) {
-				log.info("waiting for green builds " + pullRequest);
-			} else if (rebaseNeeded(pullRequest)) {
-				rebaseService.rebase(pullRequest);
-			} else if (!isApproved(pullRequest)) {
-				log.warn("approval required " + pullRequest);
-			} else {
-				merge(pullRequest);
-			}
-		}
+
+		config.getRepos().forEach(repo -> {
```

Comment:
Personally, I don't like nested lambdas. I think I would prefer traditional `for` loops.